apps/meteor/ee/server/local-services/omnichannel/OmniEEService.ts

```diff
@@ -0,0 +1,175 @@
+import { settings } from '../../../../app/settings/server';
+import { ServiceClassInternal } from '../../../../server/sdk/types/ServiceClass';
+import { schedulerLogger } from '../../../app/livechat-enterprise/server/lib/logger';
+import { OmniJobSchedularService } from '../../sdk';
+import type { IOmniEEService } from '../../sdk/types/IOmniEEService';
+import { OMNI_JOB_NAME } from './OmniJobSchedularService';
+
+type OmniOnHoldJobData = {
+	roomId: IRoom['_id'];
+	comment: string;
+};
+
+type OmniAutoTransferUnansweredChatJobData = {
+	roomId: IRoom['_id'];
+};
+
+export class OmniEEService extends ServiceClassInternal implements IOmniEEService {
+	protected name = 'omni-ee-service';
+
+	private logger = schedulerLogger;
+
+	private systemUser: IUser;
+
+	constructor() {
+		super();
+		this.logger.debug('OmniEEService started');
+	}
+
+	async monitorOnHoldRoomForAutoClose(roomId: string, timeout: number, comment: string): Promise<void> {
+		this.logger.debug(`monitorOnHoldRoomForAutoClose: ${roomId}`);
+		await this.cancelMonitorOnHoldRoomForAutoClose(roomId);
+
+		const when = moment(new Date()).add(timeout, 's').toDate();
+		const job = await OmniJobSchedularService.scheduleJobAt<OmniOnHoldJobData>(OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT, when, {
+			roomId,
+			comment,
+		});
+
+		this.logger.debug(`Scheduled job: ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId} at ${job.attrs.nextRunAt}`);
+	}
+
+	async cancelMonitorOnHoldRoomForAutoClose(roomId: string): Promise<void> {
+		this.logger.debug(`cancelMonitorOnHoldRoomForAutoClose: ${roomId}`);
+
+		const totalCancelledJobs = await OmniJobSchedularService.cancelJobByRoomId(OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT, roomId);
+
+		this.logger.debug(`Unscheduled ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId} (${totalCancelledJobs} jobs cancelled)`);
+	}
+
+	async autoCloseOnHoldChat(job: Job): Promise<void> {
+		const data = job.attrs.data as OmniOnHoldJobData;
+
+		this.logger.debug(`Executing job: ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${data.roomId}`);
+
+		const { roomId, comment } = data;
+		const [room, user] = await Promise.all([LivechatRooms.findOneById(roomId), this.getSystemUser()]);
+		if (!room) {
+			this.logger.error(`Could not find room ${roomId}`);
+		}
+		const payload = {
+			user,
+			room,
+			comment,
+			options: {},
+			visitor: undefined,
+		};
+		Livechat.closeRoom(payload);
+
+		this.logger.debug(`Executed ${OMNI_JOB_NAME.AUTO_CLOSE_ON_HOLD_CHAT} for room ${roomId}`);
+	}
+
+	async monitorUnansweredRoomForAutoTransfer(roomId: string, timeout: number): Promise<void> {
+		this.logger.debug(`monitorUnansweredRoomForAutoTransfer: ${roomId}`);
+		await this.cancelMonitorUnansweredRoomForAutoTransfer(roomId);
+
+		const when = moment(new Date()).add(timeout, 's').toDate();
+
+		const [job] = await Promise.all([
+			OmniJobSchedularService.scheduleJobAt<OmniAutoTransferUnansweredChatJobData>(OMNI_JOB_NAME.AUTO_TRANSFER_UNANSWERED_CHAT, when, {
```

Comment:
If you're passsing the type as a param, why do you need the type coercion?

Reply:
You're right, it wasn't necessary. I've removed it now